client/allocrunner/taskrunner/identity_hook.go

```diff
@@ -88,39 +72,42 @@ func (h *identityHook) Prestart(context.Context, *interfaces.TaskPrestartRequest
 		return err
 	}
 
-	signedWIDs, err := h.getIdentities()
-	if err != nil {
-		return fmt.Errorf("error fetching alternate identities: %w", err)
-	}
-
+	// Start token watcher loops
 	for _, widspec := range h.task.Identities {
-		signedWID := signedWIDs[widspec.Name]
-		if signedWID == nil {
-			// The only way to hit this should be a bug as it indicates the server
-			// did not sign an identity for a task on this alloc.
-			return fmt.Errorf("missing workload identity %q", widspec.Name)
-		}
-
-		if err := h.setAltToken(widspec, signedWID.JWT); err != nil {
-			return err
-		}
+		go h.watchIdentity(widspec)
```

Comment:
Not sure I follow? Capture as in make sure it's not nil?

Reply:
Ops, sorry I should've been more clear.

Go's loop variables should only be referenced within its iteration, as it changes at every loop (this will be [fixed](https://go.dev/blog/loopvar-preview) in Go 1.22 🎉 ). I believe this will result in every goroutine watching the same `widspec`.

Reply:
oof of course 🤦‍♂️ fixed in ddce193eabc77b42a97aaee8a48e04382de4f3d8